pinecone/client.go

```diff
@@ -16,25 +18,41 @@ import (
 
 type Client struct {
 	apiKey     string
+	headers    map[string]string
 	restClient *control.Client
 	sourceTag  string
-	headers    map[string]string
+}
+
+type NewClientBaseParams struct {
+	Headers    map[string]string
+	Host       string
+	RestClient *http.Client
+	SourceTag  string
 }
 
 type NewClientParams struct {
-	ApiKey     string            // required unless Authorization header provided
-	SourceTag  string            // optional
-	Headers    map[string]string // optional
-	RestClient *http.Client      // optional
+	ApiKey     string
+	Headers    map[string]string
+	Host       string
+	RestClient *http.Client
+	SourceTag  string
 }
 
 func NewClient(in NewClientParams) (*Client, error) {
```

Comment:
I'd expect this func to just call `NewClientBase` and configure the ApiKey as an auth header

Reply:
It now ultimately calls `NewClientBase` and passes the `ApiKey` through as part of the `Headers` map.